src/file.rs

```diff
@@ -4,12 +4,19 @@
 //! Utilities for db file.
 
 use crate::{
-	error::{try_io, Error, Result},
-	parking_lot::{RwLock, RwLockUpgradableReadGuard, RwLockWriteGuard},
+	error::{try_io, Result},
+	parking_lot::RwLock,
 	table::TableId,
 };
 use std::sync::atomic::{AtomicU64, Ordering};
 
+#[cfg(not(test))]
+const RESERVE_ADDRESS_SPACE: usize = 1024 * 1024 * 1024; // 1 Gb
+
+// Use different value for tests to work around docker limits on the test machine.
+#[cfg(test)]
+const RESERVE_ADDRESS_SPACE: usize = 64 * 1024 * 1024; // 64 Mb
```

Comment:
Wondering if this could be configurable (probably these kind of limitation might exist on some cloud infra).
Maybe even a bit more granular (eg more reserved for log).

Reply:
We'll give it a round of testing on the parity infrastructure. I've tested on my arm64 vm and had no issues.